predicators/envs/grid_row.py

```diff
@@ -325,27 +337,34 @@ def simulate(self, state: State, action: Action) -> State:
         ]
         assert len(door_cells) == 1
         # Apply ddoor if we're in same cell as door
+        # Can only open door, not close
         door_cell = door_cells[0]
         robot_cell = robot_cells[0]
-        if robot_cell == door_cell and not door_open and ddoor == 1.0:
-            next_state.set(self._door, "open", 1.0)
+        if robot_cell == door_cell and not (door_target - 0.1 \
+                            <= door_open <= door_target + 0.1 \
+    and door_target1 - 0.1 <= door_open1 <= door_target1 + 0.1):
+            new_door_level = np.clip(
+                state.get(self._door, "open") + ddoor, 0.0, 1.0)
+            next_state.set(self._door, "open", new_door_level)
+            new_door1_level = np.clip(
+                state.get(self._door, "open1") + ddoor1, 0.0, 1.0)
+            next_state.set(self._door, "open1", new_door1_level)
         # Apply dlight if we're in the same cell as the light.
         assert len(robot_cells) == 1
         light_cells = [
             c for c in self._cells if self._In_holds(state, [self._light, c])
         ]
         assert len(light_cells) == 1
         light_cell = light_cells[0]
-        if robot_cell == light_cell:
-            new_light_level = np.clip(
-                state.get(self._light, "level") + dlight, 0.0, 1.0)
-            next_state.set(self._light, "level", new_light_level)
+        if robot_cell == light_cell and dlight == 0.75 and self._LightOff_holds(
+                state, [self._light]):
+            next_state.set(self._light, "level", 0.75)
```

Comment:
wait - so is it now true that the `dlight` part of the action vector is effectively useless? If that's the case, I'd prefer you just modify the sampler to be correct instead of handling the light in this way in the transition function: modifying the sampler to always succeed is more straightforward and clear.

Reply:
i see you've also modified the sampler, so why do you also need this line in the transition function?

Reply:
Wait sorry I don't quite understand, don't we still need to change the light level and to do so we first need to check that the robot is currently in the light cell, if a TurnOnLight was called (dlight == 0.75)? I guess I could technically get rid of the self._LightOff_holds(state, [self._light]), but I think we'll need to check that the light is off when we generalize to multiple lights